ci_config: remove disabled iconv

Totto16 opened the pull request:
This removes all occurrences of `libxml2:iconv=disabled` in the ci_config, as per #2048 we also can build that from scratch per wrap on windows. 

This issue was discussed over several PRs in the past

## Affected Packages

- cairo
- fontconfig
- libxml2
- libxslt


### Disabled, because cmake is used to find libxml2
See https://github.com/mesonbuild/wrapdb/pull/2051#issuecomment-2816241979

- libxmlpp

## Other unrelated errors

On Alpine: `Can't link non-PIC static library 'z' into shared library 'freetype'. Use the 'pic' option to static_library to build with PIC.`

Totto16 committed:
ci_config: remove disabled iconv

neheb commented:
hehehe no CI was ran. Hopefully all of these succeed now.

bgilbert commented on ci_config.json:
```diff
@@ -697,11 +695,10 @@
   "libxml2": {
     "_comment": [
-      "Upstream removed autodetection of iconv and other deps; see discussion on https://gitlab.gnome.org/GNOME/libxml2/-/commit/c2ccbc0fed36de1d8c595867acb5e298c9a143f6",
+      "Upstream removed autodetection of python and other deps; see discussion on https://gitlab.gnome.org/GNOME/libxml2/-/commit/c2ccbc0fed36de1d8c595867acb5e298c9a143f6",
```
The Python issue isn't about autodetection, so we can remove this line.

Totto16 commented:
> hehehe no CI was ran. Hopefully all of these succeed now.

I also thought, wow that was quick xD

Totto16 commented:
I just triggered all affected packages, by increasing their revision, that commit will be removed, but we'll know if it fails or not xD

Totto16 commented:
Well, that didn't work out that well 😂  😓

neheb commented:
```
2025-04-18T21:31:03.7667922Z libxmlpp| Run-time dependency libxml-2.0 found: NO (tried pkgconfig and cmake)
2025-04-18T21:31:03.7668997Z libxmlpp| WARNING: CMake reported that the package LibXml2 was not found, even though Meson's preliminary check succeeded.
2025-04-18T21:31:03.7669715Z 
2025-04-18T21:31:03.7670073Z subprojects/libxml++-5.4.0/meson.build:110:11: ERROR: Fatal warnings enabled, aborting
```

Totto16 commented:
> ```
> 2025-04-18T21:31:03.7667922Z libxmlpp| Run-time dependency libxml-2.0 found: NO (tried pkgconfig and cmake)
> 2025-04-18T21:31:03.7668997Z libxmlpp| WARNING: CMake reported that the package LibXml2 was not found, even though Meson's preliminary check succeeded.
> 2025-04-18T21:31:03.7669715Z 
> 2025-04-18T21:31:03.7670073Z subprojects/libxml++-5.4.0/meson.build:110:11: ERROR: Fatal warnings enabled, aborting
> ```

Ah that helpful message again, why can this happen, I never dug into the reason for that error

neheb commented:
libxmlpp upstream uses a cmake dependency thing for libxml-2.0. Doesn't really work with meson subprojects. I'd just exclude it.

bgilbert commented:
https://github.com/mesonbuild/wrapdb/pull/2054 proposes a workaround for the `CMake reported...` failures while we get a fix into Meson.

Totto16 commented:
> #2054 proposes a workaround for the `CMake reported...` failures while we get a fix into Meson.
 
So this PR is postponed until this is merged either in meson or via #2054 👍🏼

Totto16 commented:
After looking into the logs and checking each failures, the iconv fix is working, but for some of the affected projects, there are other unrelated failures (like e.g. `Can't link non-PIC static library 'z' into shared library 'freetype'. Use the 'pic' option to static_library to build with PIC.`)

These problems seem to be not caused by us but by the upstream 😓

bgilbert commented:
You can now fix the `CMake reported...` failures by rebasing.  The `non-PIC static library` failures are https://github.com/mesonbuild/meson/issues/14497 and you could work around them for testing by removing `--pre` from all jobs in `sanity_checks.yml`.

Totto16 committed:
trigger ci, this will be removed later

Totto16 committed:
wip: this commit will be remove later

it just testes the ci

remove --pre from the meson install command, to mitigate https://github.com/mesonbuild/meson/issues/14497

neheb commented:
```
2025-04-22T13:37:42.1936640Z [31mFAILED: [0msubprojects/cairo-1.18.4/util/cairo-script/libcairo-script-interpreter.2.dylib.p/cairo-script-file.c.o 
2025-04-22T13:37:42.1942290Z cc -Isubprojects/cairo-1.18.4/util/cairo-script/libcairo-script-interpreter.2.dylib.p -Isubprojects/cairo-1.18.4/util/cairo-script -I../subprojects/cairo-1.18.4/util/cairo-script -Isubprojects/cairo-1.18.4 -I../subprojects/cairo-1.18.4 -Isubprojects/cairo-1.18.4/src -I../subprojects/cairo-1.18.4/src -I/opt/homebrew/opt/libpng/include/libpng16 -I/opt/homebrew/Cellar/fontconfig/2.16.0/include -I/opt/homebrew/opt/freetype/include/freetype2 -I/opt/homebrew/Cellar/libx11/1.8.12/include -I/opt/homebrew/Cellar/xorgproto/2024.1/include -I/opt/homebrew/Cellar/libxcb/1.17.0/include -I/opt/homebrew/Cellar/libxau/1.0.12/include -I/opt/homebrew/Cellar/libxdmcp/1.1.5/include -I/opt/homebrew/Cellar/libxext/1.3.6/include -I/opt/homebrew/Cellar/libxrender/0.9.12/include -I/opt/homebrew/Cellar/pixman/0.44.2/include/pixman-1 -I/opt/homebrew/Cellar/lzo/2.10/include/lzo -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=gnu11 -O0 -g -Wmissing-declarations -Werror-implicit-function-declaration -Wpointer-arith -Wwrite-strings -Wsign-compare -Wpacked -Wswitch-enum -Wmissing-format-attribute -Wvolatile-register-var -Wstrict-aliasing=2 -Winit-self -Wno-missing-field-initializers -Wno-unused-parameter -Wno-attributes -Wno-long-long -Winline -Wno-unused-but-set-variable -Wno-enum-conversion -fno-strict-aliasing -fno-common -D_GNU_SOURCE -DCAIRO_COMPILATION -MD -MQ subprojects/cairo-1.18.4/util/cairo-script/libcairo-script-interpreter.2.dylib.p/cairo-script-file.c.o -MF subprojects/cairo-1.18.4/util/cairo-script/libcairo-script-interpreter.2.dylib.p/cairo-script-file.c.o.d -o subprojects/cairo-1.18.4/util/cairo-script/libcairo-script-interpreter.2.dylib.p/cairo-script-file.c.o -c ../subprojects/cairo-1.18.4/util/cairo-script/cairo-script-file.c
2025-04-22T13:37:42.1947760Z In file included from ../subprojects/cairo-1.18.4/util/cairo-script/cairo-script-file.c:45:
2025-04-22T13:37:42.1948640Z [1m/opt/homebrew/Cellar/lzo/2.10/include/lzo/lzo2a.h:33:10: [0m[0;1;31mfatal error: [0m[1m'lzo/lzoconf.h' file not found[0m
2025-04-22T13:37:42.1949250Z #include <lzo/lzoconf.h>
2025-04-22T13:37:42.1949660Z [0;1;32m         ^~~~~~~~~~~~~~~
```

neheb commented:
```
2025-04-22T13:36:45.0897411Z D:\a\wrapdb\wrapdb\subprojects\libxslt-1.1.43\libexslt/libexslt.h(13): fatal error C1083: Cannot open include file: 'win32config.h': No such file or directory
```

weird.

Totto16 commented:
> ```
> 2025-04-22T13:36:45.0897411Z D:\a\wrapdb\wrapdb\subprojects\libxslt-1.1.43\libexslt/libexslt.h(13): fatal error C1083: Cannot open include file: 'win32config.h': No such file or directory
> ```
> 
> weird.

Yeeah, I thought the same, when I saw that, why is this header missing, I can only imagine, that the MSVC headers have changed, and the GitHub action image ships a newer MSVC version 🤷🏼‍♂️

neheb commented:
Needs a rebase.
